### ISSUE TITLE
'undefined reference' during compilation with icc and USER-INTEL package

### ISSUE THREAD

**Issue by mkanski** [lammps/lammps#1144]
## Summary

Hi,
Recently, I tried to compile LAMMPS with icc (ver. 19.0.0.117 20180804) and Makefile.intel_cpu_openmpi, but it failed during linking (details below).

After some digging around I found a solution. I had to add additional libraries, namely:
`-lmkl_intel_ilp64 -lmkl_tbb_thread -lmkl_core -ltbb`

I'm not sure if this is actually a LAMMPS issue or a problem with my installation of Intel compilers, but I'd like to report it just in case.

## Type of Issue

Bug report

## LAMMPS Version (Bug Report)

18 Sep 2018

## Expected Behavior (Bug Report)

A correct compilation.

## Actual Behavior (Bug Report)

Compilation stops with errors like:
```
pair_dpd_intel.o: In function `void LAMMPS_NS::PairDPDIntel::eval<0, 0, 1, double, double>(int, int, LAMMPS_NS::IntelBuffers<double, double>*, LAMMPS_NS::PairDPDIntel::ForceConst<double> const&, int, int)':
pair_dpd_intel.cpp:(.text._ZN9LAMMPS_NS12PairDPDIntel4evalILi0ELi0ELi1EddEEviiPNS_12IntelBuffersIT2_T3_EERKNS0_10ForceConstIS3_EEii[_ZN9LAMMPS_NS12PairDPDIntel4evalILi0ELi0ELi1EddEEviiPNS_12IntelBuffersIT2_T3_EERKNS0_10ForceConstIS3_EEii]+0x6ff): undefined reference to `vdRngGaussian'
```

## Steps to Reproduce (Bug Report)

```
git clone https://github.com/lammps/lammps
cd lammps/src/
make yes-user-intel
make intel_cpu_openmpi
```

**Comment by stanmoore1** [comment]
`Makefile.intel_cpu_intelmpi` links in MKL for the FFTs while `Makefile.intel_cpu_openmpi` does not. Could that be the difference?

**Comment by wmbrownIntel** [comment]
Sorry, I missed that this was openmpi. Yes, the random number generator in DPD now also uses MKL. I'll move the MKL lib links away from FFT and make the files more consistent.

**Comment by akohlmey** [comment]
fixed by #1157 please confirm
